Add day of week to graph timestamps

ericcccsliu opened the pull request:
<!--
Template sourced from https://github.com/hack4impact-uiuc/life-after-hate
Shoutout to the wonderful LAH team!
-->

## Status:
:rocket: Ready

<!--
:rocket: Ready
:construction: In development
:no_entry_sign: Do not merge
-->

## Description
Adds a `weekdayFormatter`, which formats dates to include a short weekday label, and uses the `weekdayFormatter` in our long-term graph. 

<!--
A few sentences describing the overall goals of the pull request's commits.
-->

Fixes #141 

## Notes
Will need to be changed in fixing issue #147—timezone information must be added to both formatters.

## Screenshots
<img width="935" alt="Screen Shot 2023-03-30 at 12 55 38 PM" src="https://user-images.githubusercontent.com/62641231/228923678-f291e89f-cc02-4ff4-9337-b98e77f2fc1f.png">
<img width="285" alt="Screen Shot 2023-03-30 at 12 58 47 PM" src="https://user-images.githubusercontent.com/62641231/228923791-41787489-123a-42c9-a03b-38c5d8f8a656.png">


<!--
Mac OS Screenshots: ctrl + shift + cmd + 3 (entire screen) or 4 (selection of screen), then paste in editor
Mac OS GIFs: Try using Kap
Linux/Windows: Ctrl + Alt + PrintScreen (of a window) or Ctrl + Shift + PrintScreen (selection of screen), then paste in editor
-->

ericcccsliu committed:
fix post route

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
Merge branch 'main' of https://github.com/openclimatefix/pv-sites-mobile

ericcccsliu committed:
add day

AndrewLester commented on components/graphs/Graph.tsx:
```diff
@@ -1,11 +1,11 @@
 } from 'lib/graphs';
 import { useSiteData } from 'lib/hooks';
-import { FC, useState } from 'react';
+import { FC, useState, memo } from 'react';
```
Unused import
```suggestion
import { FC, useState } from 'react';
```

ericcccsliu replied:
oops sorry i was trying out some recharts custom tick label stuff that needed that

ericcccsliu committed:
merge in main to ensure functionality

 branch 'main' of https://github.com/openclimatefix/pv-sites-mobile into el/add-day-of-week-to-graph-timestamps

ericcccsliu committed:
remove unused import
